src/backend/instagram/authUrlInstagram.ts

```diff
@@ -0,0 +1,52 @@
+import { Request, ServerRoute } from '@hapi/hapi';
+import { z } from 'zod';
+
+import { configuration } from '../utilities/configuration';
+import {
+  getSecretForSession,
+  getSession,
+  PayloadWithSession,
+} from '../utilities/sessionStorage';
+import { paths } from '../endpoints/paths';
+
+import { instagramEndpoints } from './instagramEndpoints';
+
+const zodPayload = z.object({
+  sessionId: z.string(),
+});
+
+export type Input = z.infer<typeof zodPayload>;
+
+export type Output = string;
+
+async function handler(request: Request): Promise<string> {
+  const { logger } = request;
+  logger.debug('Instagram auth started');
+
+  const session = getSession(request.payload as PayloadWithSession);
+
+  const secret = getSecretForSession(session.sessionId);
+
+  const searchParams = {
+    client_id: configuration.instagram.clientId,
+    redirect_uri: instagramEndpoints.redirectUri,
+    scope: 'user_profile,user_media',
```

Comment:
Do we need access to user_media?

Reply:
Not sure, in the documentation they have mentioned both.

Reply:
I think we should try to use the minimal possible scope